pkg/pillar/types/zedmanagertypes.go

```diff
@@ -276,28 +278,160 @@ type StorageStatus struct {
+	uuidStr string, certObjStatusPtr *CertObjStatus) bool {
+	if checkCerts && ssPtr.NeedsCerts() && ssPtr.GetCertCount() != 0 {
+		if !ssPtr.CheckCertsStatusForObject(uuidStr, certObjStatusPtr) {
+			log.Infof("MaybeAddVerifyImageConfig for %s, Certs are still not ready\n",
+				safename)
+			return false
+		}
+		if !ssPtr.CheckCertsForObject() {
+			log.Infof("MaybeAddVerifyImageConfig for %s, Certs are still not installed\n",
+				safename)
+			return false
+		}
+	}
+	return true
+}
+
+// NeedsCerts certificates are required for the Storage Object
+func (ssPtr *StorageStatus) NeedsCerts() bool {
+	if len(ssPtr.ImageSignature) != 0 {
+		return true
+	}
+	return false
+}
+
+// GetCertCount returns the number of certificates for the Storage Object
+func (ssPtr *StorageStatus) GetCertCount() int {
+	cidx := 0
+	if ssPtr.SignatureKey != "" {
+		cidx++
+	}
+	if len(ssPtr.CertificateChain) != 0 {
+		for _, certURL := range ssPtr.CertificateChain {
+			if certURL == "" {
+				log.Fatalf("Empty CertURL for %s\n", ssPtr.Name)
+			}
+			cidx++
+		}
+	}
+	return cidx
+}
+
+// CheckCertsStatusForObject checks certificates for installation status
+func (ssPtr *StorageStatus) CheckCertsStatusForObject(uuidStr string,
+	certObjStatusPtr *CertObjStatus) bool {
+
+	// certificates are still not ready, for processing
+	if certObjStatusPtr == nil {
+		log.Errorf("certObj Status is still not ready for %s\n", uuidStr)
+		return false
+	}
+
+	if ssPtr.SignatureKey != "" {
+		for _, certObj := range certObjStatusPtr.StorageStatusList {
+			if certObj.Name == ssPtr.SignatureKey {
+				if certObj.Error != "" {
+					errSrc := pubsub.TypeToName(VerifyImageStatus{})
+					ssPtr.SetErrorInfo(certObj.Error, certObj.ErrorTime, errSrc)
+					return false
+				}
+				if certObj.State != DELIVERED {
+					return false
+				}
+			}
+		}
+	}
+
+	for _, certURL := range ssPtr.CertificateChain {
+		if certURL == "" {
+			log.Fatalf("Empty CertURL for %s\n", ssPtr.Name)
+		}
+		for _, certObj := range certObjStatusPtr.StorageStatusList {
+			if certObj.Name == certURL {
+				if certObj.Error != "" {
+					errSrc := pubsub.TypeToName(VerifyImageStatus{})
+					ssPtr.SetErrorInfo(certObj.Error, certObj.ErrorTime, errSrc)
+					return false
+				}
+				if certObj.State != DELIVERED {
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+// CheckCertsForObject checks availability of Certs in Disk
+func (ssPtr *StorageStatus) CheckCertsForObject() bool {
+
+	if ssPtr.SignatureKey != "" {
+		safename := UrlToSafename(ssPtr.SignatureKey, "")
+		filename := CertificateDirname + "/" + SafenameToFilename(safename)
+		if _, err := os.Stat(filename); err != nil {
+			log.Errorf("checkCertsForObject() for %s, %v\n", filename, err)
+			return false
+		}
+		// XXX check for valid or non-zero length?
+	}
+
+	for _, certURL := range ssPtr.CertificateChain {
+		if certURL == "" {
+			log.Fatalf("Empty CertURL for %s\n", ssPtr.Name)
```

Comment:
Burrying a log.Fatal in a method like this is a bad idea. Caller needs to do such things.

Reply:
Moved it back to the calling routine.